### PR TITLE
Develop a Mobile App With Ionic and Spring Boot blog post

### PR DESCRIPTION
**Pull request by mraible**
## Description:
- Add: Develop a Mobile App With Ionic and Spring Boot blog post
- Fix: Angular PWA post did not have `{% raw %}` wrapped around HTML blocks with `{{ variable }}` so code wasn't rendered

## Type of Pull Request:

- [ ] Content (Documentation updates or typo-fixes)
- [x] Blog Post
- [ ] Functional (CSS changes, JS updates, or layout modifications)

## How Has This Been Tested?
- [x] I have built this locally and verified that it does not break existing functionality.
- [ ] For functional changes, I have tested against Chrome, Firefox, and Safari, and mobile.
- [ ] For functional changes, I have added tests.

## Primary Reviewers:
- Tech review performed by @jmelberg-okta 
- QA by Michele Degges 
- [Blog Post Google Doc](https://docs.google.com/document/d/1p9B7cyDX6t2iEUPmXE0ThwRMPothgTd9yrvGwp_rIH4/edit)

### CODE REVIEW

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 1]
```diff
@@ -0,0 +1,818 @@
...
+
+    private boolean isGreat(Beer beer) {
+        return !beer.getName().equals("Budweiser") &&
+                !beer.getName().equals("Coors Light") &&
+                !beer.getName().equals("PBR");
+    }
+}
+```
+
+You should be able to start the `server` application by running it in your favorite IDE or from the command line using `mvn spring-boot:run`. If you don't have Maven installed, you can use the Maven wrapper that's included in the project (`./mvnw spring-boot:run` on *nix, `\mvnw spring-boot:run` on Windows).
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: We give installation instructions for Windows here, but neglect it throughout the remainder of the post. Are we assuming that the reader will set up the server in a Windows environment, then run the mobile app (XCode for example) on macOSX?

**tl;dr** We may want to remove the Windows guide here.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 2]
```diff
@@ -0,0 +1,818 @@
...
+```
+
+You will be prompted to select a starter project and optionally link your app to your Ionic Dashboard. For the purposes of this tutorial, choose the **tabs** starter project and do not link the app to your Ionic Dashboard. 
+
+```bash
+cd ionic-beer
+ionic serve
+```
+
+This will open your default browser on [http://localhost:8100](http://localhost:8100). You can click through the tabbed interface to see the default structure of the app.
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: This is confusing. We are specifying our `API` to `localhost:8080`, but do not see the step to change the port from `8100`?

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 3]
```diff
@@ -0,0 +1,818 @@
...
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class BeerService {
+  public API = 'http://localhost:8080';
+  public BEER_API = this.API + '/beers';
+
+  constructor(public http: Http) {}
+
+  getGoodBeers(): Observable<any> {
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Does this need to be a generic type?

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 4]
```diff
@@ -0,0 +1,818 @@
...
+import { IonicPage, NavController, NavParams } from 'ionic-angular';
+import { BeerService } from '../../providers/beer-service';
+
+@IonicPage()
+@Component({
+  selector: 'page-beer',
+  templateUrl: 'beer.html'
+})
+export class BeerPage {
+  private beers: Array<any>;
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Can this be:
```java
private beers: Array<Beer>;
```

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 5]
```diff
@@ -0,0 +1,818 @@
...
+
+You'll also need to modify `beer-service.ts` to have `get()` and `save()` methods.
+
+```typescript
+get(id: string) {
+  return this.http.get(this.BEER_API + '/' + id)
+    .map((response: Response) => response.json());
+}
+
+save(beer: any): Observable<any> {
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Shouldn't this be typesafe?

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 6]
```diff
@@ -0,0 +1,818 @@
...
+
+To deploy the app to an iPhone, start by plugging it into your computer. Then run the following commands to install ios-deploy/ios-sim, build the app, and run it on your device. 
+
+```
+npm install -g ios-deploy ios-sim
+ionic cordova build ios --prod
+open platforms/ios/MyApp.xcodeproj
+```
+
+Select your phone as the target in Xcode and click the play button to run your app. The first time you do this, Xcode may spin for a while with a "Processing symbol files" message at the top.
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: 👍  - IMO not enough tutorials out there state this. New mobile devs aren't used to Xcode's insane processing time.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 7]
```diff
@@ -0,0 +1,818 @@
...
+
+**NOTE:** If you run into code signing issues, see [Ionic's deployment documentation](http://ionicframework.com/docs/intro/deploying/#ios-devices) to see how to solve.
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: They *should* change the `Team` to `None`. No need to go through the code signing process if this isn't a production app.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 8]
```diff
@@ -0,0 +1,818 @@
...
+*************************************************************************
+```
+
+To solve this problem, open Android Studio, select “Open an existing Android Studio project” and select the `ionic-beer/platforms/android` directory. If prompted to upgrade, choose “OK”, then proceed to create a new AVD as [described in Android Studio’s documentation](https://developer.android.com/studio/run/managing-avds.html#createavd). 
+
+After performing these steps, you should be able to run `ionic cordova emulate android` and see your app running in the AVD.
+
+{% img blog/ionic-spring-boot/android-beer-list.png alt:"Android Beer List" width:"540" %}
+
+**NOTE**: If you get an application error that says "The connection to the server was unsuccessful. (file:///android/www/index.html)", add the following line to `config.xml`. This sets the default timeout to 60 seconds (default is 20). Thanks to [Stack Overflow](http://stackoverflow.com/a/31377846) for the answer.
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: To keep with the convention, lets change:
file:///android/www/index.html -> `file:///android/www/index.html`

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 9]
```diff
@@ -0,0 +1,818 @@
...
+## Learn More
+
+I hope you've enjoyed this tour of Ionic and Angular. I like how Ionic takes your web development skills up a notch and allows you to create mobile applications that look and behave natively.
+
+You can find a completed version of the application created in this blog post [on GitHub](https://github.com/oktadeveloper/spring-boot-ionic-example). If you encountered issues, please [create an issue in GitHub](https://github.com/oktadeveloper/spring-boot-ionic-example/issues/new) or hit me up on Twitter [@mraible](https://twitter.com/mraible).
+
+To learn more about Ionic and Angular, please see the following resources:
+
+* [Get started with Ionic Framework](http://ionicframework.com/getting-started/)
+* [Build Your First Progressive Web Application with Angular and Spring Boot](http://developer.okta.com/blog/2017/05/09/progressive-web-applications-with-angular-and-spring-boot)
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Use relative addressing for each of these links

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 1 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: OK, will remove Windows references.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 2 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: The API is on 8080 and the UI is on 8100. I'll add a note.

Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: On second thought, I'm not sure if a note is necessary (or what it should say). Do you think I should add one?

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 3 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: It could be `Observable` w/o a type, but tslint might complain (not sure, haven't tested).

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 4 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Yes, but most examples I've seen don't need to have modifiers on local variables. I can add one, but it might confuse users since it's still accessible by the HTML template.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 5 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: If I wanted to create a `Beer` type, yes. However, there isn't one in this example and I didn't feel like creating one. ;)

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 7 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: You do if you want to deploy it to your advice. At least in my experience with Ionic. You might be able to do a dev build, but `ionic cordova run ios` does a prod build by default.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 8 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Will fix.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 9 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Why? Is this a new policy? If so, should I do something like `../../../../blog`?

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 10]
```diff
@@ -0,0 +1,818 @@
...
+@RestController
+public class BeerController {
+    private BeerRepository repository;
+
+    public BeerController(BeerRepository repository) {
+        this.repository = repository;
+    }
+
+    @GetMapping("/good-beers")
+    @CrossOrigin(origins = "http://localhost:4200")
```
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Where is this used?

Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: This is from the previous tutorial that I ask you to clone in the beginning. https://github.com/oktadeveloper/spring-boot-angular-example/blob/master/server/src/main/java/com/example/beer/BeerController.java#L19

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 11]
```diff
@@ -0,0 +1,818 @@
...
+To deploy to Cloud Foundry, copy [this deploy.sh script](https://github.com/oktadeveloper/spring-boot-ionic-example/blob/master/deploy.sh) to your hard drive. It expects to be in a directory above your apps (e.g. `spring-boot-ionic-example`). It also expects your apps to be named `ionic-beer` and `server`.
+
+If you don't have a Cloud Foundry account, you'll need to [create one](https://account.run.pivotal.io/z/uaa/sign-up). Then install its command line tools for this script to work.
+
+```bash
+brew tap cloudfoundry/tap && brew install cf-cli
+```
+
+After logging into Cloud Foundry, you should be able to run `./deploy.sh`. This script will deploy the server and modify `beer.service.ts` to point to it. It will also try to deploy to your phone, so you may need to complete the steps below before it will work.
+
```
Comment (bdemers) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: it wasn't obvious which API server I was suppose to use for CF.  Once I found the 'tools' tab it was pretty obvious, maybe we want to include a note about that? 
cmd: `cf login -a api.run.pivotal.io`

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 12]
```diff
@@ -0,0 +1,818 @@
...
+```
+npm install -g ios-deploy ios-sim
+ionic cordova build ios --prod
+open platforms/ios/MyApp.xcodeproj
+```
+
+Select your phone as the target in Xcode and click the play button to run your app. The first time you do this, Xcode may spin for a while with a "Processing symbol files" message at the top.
+
+**NOTE:** If you run into code signing issues, see [Ionic's deployment documentation](http://ionicframework.com/docs/intro/deploying/#ios-devices) to see how to solve.
```
Comment (bdemers) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Great note!

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 13]
```diff
@@ -0,0 +1,818 @@
...
+
+To deploy the app to an iPhone, start by plugging it into your computer. Then run the following commands to install ios-deploy/ios-sim, build the app, and run it on your device. 
+
+```
+npm install -g ios-deploy ios-sim
+ionic cordova build ios --prod
+open platforms/ios/MyApp.xcodeproj
+```
+
+Select your phone as the target in Xcode and click the play button to run your app. The first time you do this, Xcode may spin for a while with a "Processing symbol files" message at the top.
```
Comment (bdemers) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Great info!

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 14]
```diff
@@ -0,0 +1,818 @@
...
+
+**NOTE:** If you run into code signing issues, see [Ionic's deployment documentation](http://ionicframework.com/docs/intro/deploying/#ios-devices) to see how to solve.
+
+Once you've configured your phone, computer, and Apple ID to work, you should be able to open the app and see the beer list you created. Below is how it looks on my iPhone 6s Plus.
```
Comment (bdemers) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: It's been a few years since I've deployed anything to an iphone. There are a bunch of rough edges gotchas here, like changing the package name, and the trust device button not having focus when it popped up.

I think overall you give people enough info to get started, but this did require me to do a little googling.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 9 above)
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: We should avoid hard links in general. Wouldn't `[Build Your First Progressive Web Application with Angular and Spring Boot](progressive-web-applications-with-angular-and-spring-boot.html)` work?

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 10 above)
Comment (jmelberg-okta) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Gotcha. Got confused 😄

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 9 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: No, that causes it to link to `http://localhost:4000/blog/2017/05/17/progressive-web-applications-with-angular-and-spring-boot.html`. Since it's not on the same day, the link 404s.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 15]
```diff
@@ -0,0 +1,818 @@
...
+
+### Android
+
+To emulate or deploy to an Android device, you'll first need to install [Android Studio](https://developer.android.com/studio/index.html). As part of the install, it will show you where it installed the Android SDK. Set this path as an ANDROID_HOME environment variable. On a Mac, it should be `~/Library/Android/sdk/`.
+
+**If you've just installed Android Studio, make sure to open it to complete the installation.**
+
+To deploy to the Android emulator, run `ionic cordova emulate android`. This will install Andorid support and print out instructions about how to create an emulator image.
+
+```
```
Comment (bdemers) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: This was a major PITA.  You did point out all of the problems I ran into though, which was awesome!  I probably would have bounced when you wanted me to install an IDE, but your instructions were pretty easy to follow

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 16]
```diff
@@ -0,0 +1,818 @@
...
+beer.html
+beer.module.ts
+beer.scss
+beer.ts
+```
+
+Open `src/app/app.module.ts` and add `BeerPageModule` to the `imports` list in `app.module.ts`.
+
+```typescript
+import { BeerPageModule } from '../pages/beer/beer.module';
```
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: No, our current syntax highlighting doesn't seem to support that.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 17]
```diff
@@ -0,0 +1,818 @@
...
+<ion-header>
+  <ion-navbar>
+    <ion-title>Good Beers</ion-title>
+    <ion-buttons end>
+      <button ion-button icon-only (click)="openModal()" color="primary">
+        <ion-icon name="add-circle"></ion-icon>
+        <ion-icon name="beer"></ion-icon>
+      </button>
+    </ion-buttons>
+  </ion-navbar>
```
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Done.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md [hunk 18]
```diff
@@ -0,0 +1,818 @@
...
+
+At this point, if you try to add or edit a beer name, you'll likely see an error in your browser's console.
+
+```
+Uncaught (in promise): Error: No component factory found for BeerModalPage. 
+Did you add it to @NgModule.entryComponents?
+```
+
+To fix this problem, add `BeerModalPage` to the `declarations` and `entryComponent` lists in `beer.module.ts`.
+
```
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Done.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 15 above)
Comment (mraible) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: Yeah, getting it working on Android was a lot harder.

File: _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md (hunk 16 above)
Comment (bdemers) [inline _source/_posts/2017-05-17-develop-a-mobile-app-with-ionic-and-spring-boot.md]: i figured
